[NativeRT] Make VariadicOpConverter and FuseListUnpackConverter for cpu nodes only

SherlockNoMad opened the pull request:
Summary:
VariadicOpConverter and FuseListUnpackConverter would introduce ops that only have CPU kernels.

Currently, the graph passes are ran if static_dispatch is enabled.

As we plan to enable static_dispatch by default, this diff add the additional check for the graph pass to only work on the node that has all the inputs/outputs on CPU.

Test Plan:
CI

Rollback Plan:

Differential Revision: D79295640

henryoier commented on torch/nativert/graph/GraphUtils.cpp:
```diff
@@ -0,0 +1,76 @@
+
+namespace torch::nativert {
+
+bool allIOTensorsAttributesAreOnCpu(const Node& node) {
```
Nit: areAllIOTensorsAttributesOnCpu

henryoier commented on torch/nativert/graph/GraphUtils.h:
```diff
@@ -0,0 +1,22 @@
+ * @return true if all I/O tensors and device attributes are on CPU, false
+ * otherwise
+ */
+bool allIOTensorsAttributesAreOnCpu(const Node& node);
```
Ditto

henryoier commented on torch/nativert/graph/GraphUtils.cpp:
```diff
@@ -0,0 +1,76 @@
+          }
+        }
+      }
+    }
```
Ditto

henryoier commented on torch/nativert/graph/GraphUtils.cpp:
```diff
@@ -0,0 +1,76 @@
+            return false;
+          }
+        }
+      }
```
Add a else and logs here if the value type is unsupported or invalid?

SherlockNoMad committed:
[NativeRT] Make VariadicOpConverter and FuseListUnpackConverter for cpu nodes only (#159519)

Summary:

VariadicOpConverter and FuseListUnpackConverter would introduce ops that only have CPU kernels. 

Currently, the graph passes are ran if static_dispatch is enabled. 

As we plan to enable static_dispatch by default, this diff change the graph pass to only work on the node that has all the inputs/outputs on CPU.

Test Plan:
CI

Rollback Plan:

Reviewed By: dolpm, georgiaphillips, henryoier

Differential Revision: D79295640
